### PR TITLE
Updated release procedure and dev env docs

### PR DESCRIPTION
**Pull request by stchris**
To make the documentation more inline with what we have in `followthemoney`:

- I updated the README with local dev env and release procedure sections
- Moved the dev requirements to a `requirements-dev.txt` file

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,6 +50,3 @@ clean:
 	find . -name '*.pyc' -exec rm -f {} +
 	find . -name '*.pyo' -exec rm -f {} +
 	find . -type d -name __pycache__ -exec rm -r {} \+
-
-dev: 
-	pip install -q bump2version
```
Comment (Rosencrantz) [inline Makefile:55]: This is because bump2version isn't used as part of a dev environment?

Comment (stchris) [reply Makefile:55]: I just moved it to `requirements-dev.txt` to be consistent with what we do in other repos.

Comment (stchris) [reply Makefile:55]: Unless `make dev` is a thing we do in other places, then I'm happy to adjust that.

**Review by Rosencrantz — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (Rosencrantz) [reply Makefile:55]: No, cool. Makes sense

**Pull request merged**